docusaurus/docs/reactnative/guides/performance_boost.mdx

```diff
@@ -0,0 +1,91 @@
+chat application if you are expecting heavy traffic on your chat.
+
+## Channel Settings
+
+Within each channel type, some settings are available that apply to all channels of that type.
+Among these settings are the ability to enable/disable each event type.
+When disabled, events of that type (for channels of this type) will not be passed through to a client's open WebSocket connection.
+It's also important to note that increasing the events enabled on a channel type also increases the load on clients in those channels.
+
+- For Livestream type channels, we recommend disabling Connect, Read Events, and Typing Events. These will cause performance issues and don't generally add to the user experience in these use cases.
+- Also consider using [Slow Mode](https://getstream.io/chat/docs/slow_mode/?language=js#channel-slow-mode) for Livestream events.
+
+<img src={channelSettings} />
+
+## State update throttling
+
+[Channel](../core-components/channel.mdx) component updates its internal react state based on chat related events received from websocket.
+By default these state updates are throttled to once per 500 ms. For high traffic applications, you may want to adjust throttling
+interval to a higher number. You can configure throttle interval via props on Channel component:
+
+- `newMessageStateUpdateThrottleInterval` prop adjusts the throttling interval for state updates when new messages arrive in chat.
+- `stateUpdateThrottleInterval` prop adjusts the throttling interval for state updates for all the other updates to channel except new message.
+
+We recommend following configuration, but you can adjust it as per your requirement.
+
+```tsx
+<Channel
+  newMessageStateUpdateThrottleInterval={2000}
+  stateUpdateThrottleInterval={800}
+/>
+```
+
+:::note
+- These props are available in version >= v3.9.0
+- Changes to `stateUpdateThrottleInterval` will result in delays in adding a reaction to message. So try not to keep too high value for this prop.
+:::
+
+## Heavy UI features
+
+You can relieve the JS thread by disabling some of the heavy features as mentioned below.
+
+:::note
+These props are available in version >= v3.9.0
+:::
+
+### Image Viewer
+
+By default, when you open (or press on) an image attachment, it opens the image in full screen image viewer.
+And within the viewer, you can keep swiping left to scroll through all the images within channel (which have been loaded so far).
+This creates quite a lot of extra work for JS thread in terms of keeping track of image attachments loaded in channel and pre-populating them in the viewer
+for smooth transitions.
+
+Since v3.9.0 we have introduced an alternate UX for image viewer, where image attachments of only the selected messages will be loaded in the viewer.
+You can enable this alternate UX by setting a prop `legacyImageViewerSwipeBehaviour` as false on Channel component.
```

Comment:
```suggestion
By default, when you open (press) an image attachment it opens the image in a full-screen image viewer.
Within the viewer, you can keep swiping left to scroll through all the images that have been loaded within a channel so far.
This creates quite a lot of extra work for the JS thread to keep track of image attachments loaded in a channel and pre-populating them in the viewer for smooth transitions.

In v3.9.0, we introduced an alternate UX for the image viewer where image attachments are only loaded for the selected messages in a viewer.
You can enable this alternate UX by setting the prop `legacyImageViewerSwipeBehaviour` to false on a Channel component.
```

For the selected messages? From the selected messages? Not entirely sure what would be correct here